monkey/infection_monkey/puppet/puppet.py

```diff
@@ -57,9 +60,10 @@ def fingerprint(
         return fingerprinter.get_host_fingerprint(host, ping_scan_data, port_scan_data, options)
 
     def exploit_host(
-        self, name: str, host: str, options: Dict, interrupt: threading.Event
+        self, name: str, host: object, options: Dict, interrupt: threading.Event
     ) -> ExploiterResultData:
-        return self._mock_puppet.exploit_host(name, host, options, interrupt)
+        exploiter = self._plugin_registry.get_plugin(name, PluginType.EXPLOITER)
+        return exploiter(host, LegacyTelemetryMessengerAdapter(), options).exploit_host()
```

Comment:
Parameters should be passed to `exploit_host`, not to the constructor. The object shouldn't be constructed here.

Reply:
See https://github.com/guardicore/monkey/pull/1727#discussion_r811915428